gef.py

```diff
@@ -4308,9 +4308,42 @@ def do_invoke(self, argv):
         new_output = argv[1]
 
         disable_context()
-        res = gdb.execute("""call open("{:s}", 66, 0666)""".format(new_output), to_string=True)
-        # Output example: $1 = 3
-        new_fd = int(res.split()[2], 0)
+
+        if ':' in new_output:
+            address = socket.gethostbyname(new_output.split(':')[0])
+            port = int(new_output.split(':')[1])
+
+            # AF_INET = 2, SOCK_STREAM = 1
+            res = gdb.execute("""call socket(2, 1, 0)""", to_string=True)
+            new_fd = int(res.split()[2], 0)
+
+            # fill in memory with sockaddr_in struct contents
+            # we will do this in the stack, since connect() wants a pointer to a struct
+            vmmap = get_process_maps()
+            stack_addr = next(entry.page_start for entry in vmmap if entry.path == '[stack]')
```

Comment:
Why use a generator here? Why not a list comprehension and grab index 0? Seems easier to read.